sktime/annotation/stray.py

```diff
@@ -0,0 +1,283 @@
+        self.p = p
+        self.size_threshold = size_threshold
+        self.outlier_tail = outlier_tail
+        super(STRAY, self).__init__()
+
+    def _find_threshold(self, outlier_score, n):
+        """Find Outlier Threshold.
+
+        Parameters
+        ----------
+        outlier_score : np.ArrayLike
+            The outlier scores determined by k nearast neighbours distance
+        n : int
+            The number of rows remaining in X when NA's are removed.
+
+        Returns
+        -------
+        array of indices of the observations determined to be outliers.
+        """
+        if self.outlier_tail == "min":
+            outlier_score = -outlier_score
+
+        order = np.argsort(outlier_score)
+        gaps = np.append(0, np.diff(outlier_score[order]))
+        n4 = int(max(min(self.size_threshold, np.floor(n / 4)), 2))
+
+        J = np.array([i for i in range(2, n4 + 1)])
+        start = int(max(np.floor(n * (1 - self.p)), 1))
+
+        ghat = [
+            0.0 if i < start else sum((J / (n4 - 1)) * gaps[i - J + 1])
+            for i in range(n)
+        ]
+
+        log_alpha = np.log(1 / self.alpha)
+        bound = np.Inf
+
+        for i in range(start, n):
+            if gaps[i] > log_alpha * ghat[i]:
+                bound = outlier_score[order][i - 1]
+                break
+
+        return np.where(outlier_score > bound)[0]
+
+    def _use_KNN(self, X, n):
+        """Find outliers using kNN distance with maximum gap.
+
+        Parameters
+        ----------
+        X : np.ArrayLike
+            Data for anomaly detection (time series).
+        n : int
+            The number of rows remaining in X when NA's are removed.
+
+        Returns
+        -------
+        dict of index of outliers and the outlier scores
+        """
+        if len(X.shape) == 1:
+            nbrs = NearestNeighbors(n_neighbors=self.k + 1).fit(X.reshape(-1, 1))
+            distances, _ = nbrs.kneighbors(X.reshape(-1, 1))
+        else:
+            nbrs = NearestNeighbors(
+                n_neighbors=n if self.k >= n else self.k + 1,
+                algorithm=self.knn_algorithm,
+            ).fit(X)
+            distances, _ = nbrs.kneighbors(X)
+
+        if self.k == 1:
+            d = distances[:, 1]
+        else:
+            diff = np.apply_along_axis(np.diff, 1, distances)
+            d = distances[range(n), np.apply_along_axis(np.argmax, 1, diff) + 1]
+
+        out_index = self._find_threshold(d, n)
+        return {"idx_outliers": out_index, "out_scores": d}
+
+    def _find_HDoutliers(self, X):
+        """Detect Anomalies in High Dimensional Data.
+
+        Parameters
+        ----------
+        X : np.ArrayLike
+            Data for anomaly detection (time series).
+
+        Returns
+        -------
+        dict of anomalies and their corresponding scores
+        """
+        r = np.shape(X)[0]
+        idx_dropna = np.array([i for i in range(r) if not np.isnan(X[i]).any()])  # tag
```

Comment:
how about [np.isnan](https://numpy.org/doc/stable/reference/generated/numpy.isnan.html) instead?

Reply:
ahh, yeah - changed it to `idx_dropna = np.argwhere((~np.isnan(X)).all(axis=1)).flatten()` instead

Reply:
damn, changing back because `AxisError: axis 1 is out of bounds for array of dimension 1` in 1D case